### PR TITLE
[sw] Remove |strlen()| from our common libraries.

### PR DESCRIPTION
**Pull request by mcy**
We've already made the decision to avoid NUL-termination APIs in device code; this change simply enforces it.

### CODE REVIEW

File: sw/device/tests/hmac/sha256_test.c [hunk 1]
```diff
@@ -7,42 +7,38 @@
 #include "sw/device/lib/hw_sha256.h"
 #include "sw/device/lib/uart.h"
 
-typedef struct test_data {
-  uint32_t digest[8];
-  char data[512];
-} test_data_t;
+const size_t kDataLen = 142;
+const char kData[142] =
```
Comment (gkelly) [inline sw/device/tests/hmac/sha256_test.c]: `static`

Comment (mcy) [reply sw/device/tests/hmac/sha256_test.c]: Got it.

**Comment by gkelly** [comment]
```
make: *** No rule to make target '../build/sim/boot_rom/lib/strlen.o', needed by '../build/sim/boot_rom/lib/libot.a'.  Stop.
```

**Comment by mcy** [comment]
I'm going to just block on the Make cutover.

**Pull request merged**